diversity_analysis_tool/diversity.py

```diff
@@ -161,8 +211,23 @@ def create_age_bands(original_df, age_field_name, start_age=0, end_age=90, years
     )
 
     df[banded_field_name] = df[banded_field_name].astype(str)
-    df[banded_field_name] = df[banded_field_name].apply(lambda x: str(x[x.find("(") + 1:x.find("]")]))
-    df[banded_field_name] = df[banded_field_name].apply(lambda x: x.replace(',', ' -'))
+    df[banded_field_name] = df[banded_field_name].apply(
+        lambda x: str(x[x.find("(") + 1 : x.find("]")])
+    )
+    df[banded_field_name] = df[banded_field_name].apply(lambda x: x.replace(",", " -"))
+
+    # the bands need to be ordered by first age in range for visualisation later
+    order = df[banded_field_name].sort_values().unique()
+    order = [x for x in order if x != "na"]
+    sorted_idx = np.argsort([int(x[:2]) for x in order])
+    new_order = [order[i] for i in sorted_idx]
```

Comment:
I think we can do first the `unique()` and then `sort_values()` so we just sort the unique values instead of the full list :) 

Also, I think if `order` is a numpy array we might be able to call it directly instead of having to iterate through the list again (this one might be picky, sorry!)

```
order = df[df[banded_field_name]!='na'][banded_field_name].unique()
order = np.sort(order)
sorted_idx = np.argsort([int(x[:2]) for x in order])
new_order = order[sorted_idx]
```

Reply:
Check this works properly, just in case :P

Reply:
yup, good call that works! in fact, we don't even need to sort the order before making the new order :)